### PR TITLE
config: add support for logger provider configuration

### PR DESCRIPTION
**Pull request by codeboten**

### CODE REVIEW

File: config/log.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return noop.NewLoggerProvider(), noopShutdown, errors.Join(errs...)
+	}
+
+	mp := sdklog.NewLoggerProvider(opts...)
+	return mp, mp.Shutdown, nil
```
Comment (MrAlias) [inline config/log.go:42]: nit

```suggestion
	lp := sdklog.NewLoggerProvider(opts...)
	return lp, lp.Shutdown, nil
```

File: config/log_test.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantErr: errors.New("unsupported compression \"invalid\""),
+		},
+		{
+			name: "simple/no-exporter",
```
Comment (MrAlias) [inline config/log_test.go:313]: Missing test for simple with a valid exporter.

File: config/log.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		return batchLogProcessor(processor.Batch, exp)
+	}
+	if processor.Simple != nil {
+		exp, err := logExporter(ctx, processor.Simple.Exporter)
+		if err != nil {
+			return nil, err
+		}
+		return sdklog.NewSimpleProcessor(exp), nil
+	}
+	return nil, fmt.Errorf("unsupported log processor type %v", processor)
```
Comment (MrAlias) [inline config/log.go:63]:
```suggestion
	return nil, fmt.Errorf("unsupported log processor type %[1]#v (%[1]T)", processor)
```

File: config/log.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+func logExporter(ctx context.Context, exporter LogRecordExporter) (sdklog.Exporter, error) {
+	if exporter.OTLP != nil {
+		switch exporter.OTLP.Protocol {
+		case protocolProtobufHTTP:
+			return otlpHTTPLogExporter(ctx, exporter.OTLP)
+		default:
+			return nil, fmt.Errorf("unsupported protocol %q", exporter.OTLP.Protocol)
```
Comment (MrAlias) [inline config/log.go:72]: Missing test case.

File: config/log.go (hunk 3 above)
Comment (codeboten) [reply config/log.go:63]: did you mean to leave both v and T in here? with both it ends up logging something like:

```
"unsupported log processor type {<nil> <nil> <nil>} (config.LogRecordProcessor)"
```

File: config/log_test.go (hunk 2 above)
Comment (codeboten) [reply config/log_test.go:313]: Added

File: config/log.go (hunk 4 above)
Comment (codeboten) [reply config/log.go:72]: test added

File: config/log.go (hunk 3 above)
Comment (MrAlias) [reply config/log.go:63]: Yeah, I was mostly thinking we need to include the type to support the rest of the message. If including the instance value as well isn't going to be helpful I'm fine removing that.

Comment (codeboten) [reply config/log.go]: i updated the error message to specify the type of processors that are expected instead. the type of processor isn't support helpful and neither are the null values

**Review by MrAlias — APPROVED** [review]

**Comment by codeboten** [comment]
@MrAlias is there anything else on this PR that's needed to move it forward?

**Comment by MrAlias** [comment]
It needs another review from a non-Splunker.

cc @open-telemetry/go-approvers

**Review by dashpole — APPROVED** [review]

**Comment by codeboten** [comment]
Thanks @MrAlias @dashpole!

**Pull request merged**